engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -258,15 +260,33 @@ - (instancetype)init {
 
 - (void)sharedSetupWithProject:(nullable FlutterDartProject*)project
                   initialRoute:(nullable NSString*)initialRoute {
-  // Need the project to get settings for the view. Initializing it here means
-  // the Engine class won't initialize it later.
-  if (!project) {
-    project = [[FlutterDartProject alloc] init];
-  }
-  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"io.flutter"
-                                                      project:project
-                                       allowHeadlessExecution:self.engineAllowHeadlessExecution
-                                           restorationEnabled:self.restorationIdentifier != nil];
+  id appDelegate = FlutterSharedApplication.application.delegate;
```

Comment:
sanity check - this part is unchanged right?

Reply:
(I've maintained the old commit so you can see what has been added since the revert)

The only thing changed in this method is the call to the FlutterAppDelegate.pluginRegistrant.